### DISCUSSION TITLE
Does PDML have a logo?

### DISCUSSION THREAD

**Discussion by slanden** [pdml-lang/pdml-lang.github.io#8] [Q&A] [answered]
I started working on [language support for VS Code](https://marketplace.visualstudio.com/items?itemName=shelby-landen.pdml) and a logo would look nice on the Marketplace page. If you don't have one already I can think about this some more and brainstorm ideas here if you'd like.

Side note: if you check out the linked extension's Marketplace page and see any issues with the Readme file, please let me know. The introductory summary is also not finished.

**Comment by pml-lang** [comment, marked as answer]
> I started working on language support for VS Code

AWESOME !!! Thank you.

> a logo would look nice on the Marketplace page

Yes!

>  If you don't have one already

No. It's on my to-do list.

>  I can think about this some more and brainstorm ideas here if you'd like.

That would be great.

> if you check out the linked extension's Marketplace page and see any issues with the Readme file

I didn't see any issues.

I tried it out. The installation in my local VSCode instance went flawlessly (searched for 'pdml' in 'Extensions Marketplace'), and I found the WIP to be very useful already.

Maybe `pml` could be added as an additional file extension (besides `pdml`) in [package.json](https://github.com/slanden/vscode-pdml/blob/main/package.json), as the plugin is already useful for PML documents too.

If you agree, I will add a link to your plugin on the PDML and PML websites.

**Reply by slanden** [reply]
> AWESOME !!! Thank you.

Thanks for PDML!

> I found the WIP to be very useful already.

Nice! Thanks for trying it out.

> Maybe pml could be added as an additional file extension

So, my only reason for starting without it was because I want to be sure that `.pdml` files only get PDML functionality while specialized file formats get PDML + whatever the specialization has. I say it that way because in addition to PML, I plan to also support HTML using PDML syntax. They would both get their own auto-completions, validations, and whatever else.  I think VS Code can do this, I just wanted to confirm that and then I'll add it.

> If you agree

Please feel free to do so, and thanks!

**Reply by pml-lang** [reply]
> my only reason for starting without it was because I want to be sure that .pdml files only get PDML functionality while specialized file formats get PDML + whatever the specialization has.

Yes. That makes perfect sense!

Note: Yesterday I bought the domain `pdml-lang.dev` (to complement with the existing sites `pml-lang.dev` and  `ppl-lang.dev`). I'll let you know when the new site is ready.

**Reply by slanden** [reply]
I have a path forward for PML support, so I enabled the ".pml" file extension. No PML-specific functionality yet though.

As I was thinking more about the extension's future for supporting the additional features for PML and HTML-using-PDML-syntax, I was reading up on the [PDML license](https://creativecommons.org/licenses/by-nd/4.0/). I want to make sure I don't violate it. Would the _HTML-like_ violate the derivatives restriction?

What about, for example, if a Golf company has their own vocabulary for scoring golf games ([I found such a thing](https://code.google.com/archive/p/golfml/) in my original search for markup languages) they want to implement for their company or something. Would it be a derivative, and hence violation, to use PDML for their golf vocabulary and how their custom nodes can interact with each other (e.g. rules like _ListItem_ nodes can only be nested inside _List_ nodes)?

I feel like I'm misinterpreting the license, considering you said that PML is to PDML like HTML is to XML, but I'd rather be sure.

> I'll let you know when the new site is ready.

Awesome, thanks!

**Reply by pml-lang** [reply]
> I enabled the ".pml" file extension

Nice. Thanks.

> Would it be a derivative, and hence violation, to use PDML for their golf vocabulary and how their custom nodes can interact with each other (e.g. rules like ListItem nodes can only be nested inside List nodes)?

No, that would not be a violation.
PDML is a general-purpose language, and people can define their custom dialects (similar to PML) with domain-specific nodes and validation rules.
Hence, you could also use PDML to create an HTML-like language.

**Reply by slanden** [reply]
> people can define their custom dialects (similar to PML) with domain-specific nodes and validation rules.

Great, thanks for clearing that up.

So, I was doodling [some ideas for a logo](https://drive.google.com/file/d/1q-Y_OMCZoeVwm671r9mjGLd19_LvyHfm/view?usp=sharing)... Just some thoughts; I'm no designer and I didn't put much thought into the colors, I just used what I found on the PDML and PML websites. Again, just brainstorming.

Paying a logo designer from Fiverr.com is not off the table either just to have something, I've paid like $20-30 in the past for other logos. I'm not rushing you, I'm just saying I could look into that if you're open to it or if you'd rather go a different route of course that's cool too.

**Reply by pdml-lang** [reply]
> I was doodling some ideas for a logo.

Great! I like the idea to use the square brackets `[ ]` in the logo, and to use similar logos for PDML 'dialects'.

> Paying a logo designer from Fiverr.com is not off the table

Yes, good idea! I'm not a designer too.

> I could look into that if you're open to it

Yes, please! Very much appreciated!

I will also ponder about it and post an additional comment.

**Reply by pdml-lang** [reply]
A few thoughts (just brainstorming):

- The logo should be available as an SVG file, as well as PNG files with commonly used resolutions.

- There should be no sharp corners. Only rounded corners, because they are more pleasant to look at.
    
    ![corners](https://user-images.githubusercontent.com/93634328/195496163-0d4bf7a5-166a-4484-9813-86c67ea1d697.png)

- As I said already, I like your idea to use the square brackets [ ] in the logo, and to use similar logos for PDML 'dialects'.
    For PDML we could use:
    ```
    [dm]
    ```
    `dm` would stand for 'data and markup'.
    And for PML we could use:
    ```
    [m]
    ```
    ... where the `m` stands for 'markup'.
    For other dialects, just the letters between the brackets would change (e.g. `[h]` for an HTML-like dialect).

- For the PDML logo we could use red green and blue colors. Reason: These are the basic colors to create any other color, in the same way as PDML is the basic format to create other formats. For example: red for the left bracket, green for `dm`, and blue for the right bracket.

These are just some quick ideas. I'm open for any other suggestions.

**Reply by slanden** [reply]
The designer delivered her designs, I've shared them with you but [here's the public link](https://www.fiverr.com/collaborate/eyJsaW5rX2lkIjoiNjM0YWRiZjM4NjhjMWEwMDEwYjYyZTcxIiwidHlwZSI6InNoYXJlYWJsZS1saW5rIn0=) just in case you didn't get it or anyone else would like to comment on any of them.

Here are my thoughts on them...

**1. Logo with red page and white "[PDML]" label**

This looks like an icon you might see by default on your operating system; Looks good, but not unique. Although, it's for a language so.. I could be over thinking it.

**2. Logo with red page with "[ ]" on the right side**

The page and label have a similarly default look as I mentioned in the other red-page one, but this one looks more unique with the "[ ]" on the outside. It has more emphasis. If other "dialects" are incorporated, I think the page color could change and keep the brackets the same color since those are the constant.

**3. Logo with the "[ ]" in a red box with "PDML" outside of it**

I might like this one the best, but am not sure other "dialects" would work. Maybe the color of the box would change for each dialect.

**4. Logo with red, green, blue**

This one looks better than I thought it would. I do get a sense that it might be a little too busy though; Three colors are a lot for such a simple logo.

If each "dialect" has its own color, there are two additional colors it would have to look good with instead of one.

It looks more energetic than some of the others.

**Reply by pml-lang** [reply]
Nice!

My thoughts:

- The text should be inside the brackets, because that's a basic principle in PDML.

- Using PDML (four letters) feels a bit 'too much letters'. Just using D and M might be enough for the logo, and reflects the essence of PDML: data and markup.

- Not sure about the idea of adding a page to the logo. It feels unnecessary to me.

- I would like to see a variation of the logo with lowercase letters (e.g. `[dm]`) and a font that has only rounded corners. For example, I think that the 'd' and 'm' as designed in [Modulus](https://creativemarket.com/AndrewFootit/269252-Modulus) (not a free font) look very nice and modern. Similar letters could also easily be created with vectors in SVG. More ideas for rounded fonts are shown at [30 Rounded Fonts that Add Modern Simplicity](https://creativemarket.com/blog/smooth-rounded-fonts).

- For the logo using three colors: The order of the colors should be RGB (i.e. left bracket in red, text in green, right bracket in blue). It would be interesting to see a variation using only pure colors (e.g. no red or blue components in green), even though I think that the colors in the examples look nicer. I agree that "Three colors are a lot for such a simple logo", therefore another idea would be to use only a touch of RGB (slightly red, green, and blue).

- Last suggestion: Brackets bigger than text, and text vertically centered inside the brackets, to avoid making the logo look "just like text".

My suggestions in a nutshell:

- `[dm]`

- rounded font (like [Modulus](https://creativemarket.com/AndrewFootit/269252-Modulus))

- brackets bigger than text, and text centered vertically

- pure RGB, or just a touch of RGB, or just a single color

Questions: Will we be the owner of the logo, and have all rights to create derivatives? Will the logo be delivered in SVG format?

PS: I didn't comment on Fiverr. IMO it's better for the designer to communicate with only one person. But if you prefer, I'll post a comment on Fiverr.

**Reply by slanden** [reply]
> Using PDML (four letters) feels a bit 'too much letters'

I get where you're coming from. Gotta love those three letter acronyms when you can get 'em. On the other hand though, using just "dm" feels almost like it's no longer a name and more like a category, "Data and Markup". I think "Practical" is an important identifying part.

However, I think lowercase letters as well as some fonts could help make it look.. slimmer.

<details>
<summary>See if you like any of these free fonts</summary>
<br>

Fabada (this is used in the file shared below)
https://www.1001fonts.com/fabada-font.html?text=%5Bpdml%5D&commercial=true

TimeBurner
https://www.fontspace.com/timeburner-font-f15111

Maxwell (this is used in the file shared below)
https://www.1001fonts.com/maxwell-font.html?text=%5Bpdml%5D&commercial=true

Comfortaa
https://www.fontspace.com/comfortaa-font-f8306

Plymdale
https://www.fontspace.com/plymdale-font-f7369

Yaahowu
https://www.1001fonts.com/yaahowu-font.html?text=%5Bpdml%5D&commercial=true

Glory
https://fonts.google.com/specimen/Glory/tester?category=Sans+Serif,Display,Monospace&preview.text=%5Bpdml%5D&preview.text_type=custom

Gugi
https://fonts.google.com/specimen/Gugi/tester?category=Sans+Serif,Display,Monospace&preview.text=%5Bpdml%5D&preview.text_type=custom

Lobster
https://fonts.google.com/specimen/Lobster/tester?category=Sans+Serif,Display,Monospace&preview.text=%5Bpdml%5D&preview.text_type=custom

</details>

I updated the original logo ideas file I shared, but it was doing something funky with the logos so I uploaded a [separate PDF version](https://drive.google.com/file/d/1Y9wsQb83A6uyryUJM7cYM-5QOT8MDTyD/view?usp=sharing).

It includes variations with two new fonts, and a set using RGB. There's also one with pure RGB. I have to say though, I find the pure RGB one hard to look at.


We have full rights over the logos, once we settle on something I'll share the final version.

**Reply by pml-lang** [reply]
Thank you. I like your first four examples.

## Color

I now think we should abandon my idea of using RGB colors, because that looks a bit cheesy and aggressive. The green color in your examples looks very nice and peaceful. I like it. So let's stick with that color.

## Font

Faba and Maxell (as used in your examples) with lowercase letters both look very good. I would suggest to use Maxell.

## Text

> using just "dm" feels almost like it's no longer a name and more like a category, "Data and Markup". I think "Practical" is an important identifying part.

IMO a logo does not need to show the name - it's just a logo, an abstract picture.
I still think that four letters is too much. I was even thinking about merging the letters `d` and `m`. It would look like this (just a 'quick and dirty' test):

![pdml_logo](https://user-images.githubusercontent.com/77769154/197090284-57c59dea-6d79-48d9-ad0b-c9e54875d1d3.png)

## Width / Height Ratio

IMO the logo looks best if it's either a square or a  [golden ratio](https://en.wikipedia.org/wiki/Golden_ratio) rectangle (1.618/1) with slightly rounded edges.

PS: We have to agree on how I'll pay the designer. I don't have your email address. I suggest you send me a mail to chris @ pml-lang dot dev to discuss this.

**Reply by slanden** [reply]
> IMO a logo does not need to show the name - it's just a logo, an abstract picture.

I generally agree. Predominantly text logos just switch my brain from looking at an abstract symbol to just reading the text as if it were the name/title. But anyway, I liked the idea of merging the letters. I didn't try the letters without merging them yet though.

Check out that G Drive link again to see if you like any of the updated variations. I made the text darker for more contrast, but I still have the first text color in case we need to change it back.

> IMO the logo looks best if it's either a square or a [golden ratio](https://en.wikipedia.org/wiki/Golden_ratio) rectangle

I kept this in mind too. Maybe a logo designer could correct me if I'm wrong, but from what I've seen it's not the logo as a whole but the pieces of the whole that are created using the golden ratio. It would make sense because I tried making the pieces and the whole use the ratio, but the text was either puny or touching the brackets and didn't look good.

> how I'll pay the designer

Thanks, but don't worry about it. I paid her already and I intended to all along. It wasn't much, I just wanted a logo for the extension so win-win.

Whenever it's finished though, should I upload it to the repo or email it to you?

**Reply by pml-lang** [reply]
> Check out that G Drive link again to see if you like any of the updated variations.

Thanks. The upper left version looks really nice.

> I paid her already

That's very generous of you! Thank you so much.

> should I upload it to the repo or email it to you?

Whatever you prefer.

**Reply by slanden** [reply]
> Thanks. The upper left version looks really nice.

I like it too, that was the first I made. The only issue I see is it looks like 'd' and 'n'... What do you think?
Are you still happy with the colors?

You're welcome!

**Reply by pml-lang** [reply]
> The only issue I see is it looks like 'd' and 'n'... What do you think?

I didn't see an 'n', but that's maybe just because my brain is already "too much programmed" to see an 'm'. But for some people it could indeed look like an 'n'. Now I understand why you suggested the alternative upper right version. However, IMO that version looks a bit confusing. Maybe the ambiguity can be eliminated (or at least mitigated) by merging the _full_ 'm', as shown below (look at the lower right of the 'd' which is different from your example):

![dm_50](https://user-images.githubusercontent.com/77769154/197437698-35bedbae-feb1-4f12-94f8-edc54ea9ecce.png)

> Are you still happy with the colors?

I'm not sure about using two colors.
IMO the color of the brackets looks better - more "brilliant and refreshing". On the other hand, two colors makes the logo "less boring".
Maybe it would be interesting to compare two versions side by side to take a final decision (one version using only the color of the brackets).

**Reply by slanden** [reply]
Alright at the same link I added a version like your sketch, and both now have a single-color variant.

A 'd' usually has that little piece at the bottom anyway, so I'm not sure how much it differentiates the 'd' from the 'm'. In other words, it might still just look like part of the 'd' instead of part of the 'm'. If you think it helps, we can go with that. Otherwise, it might not be a big enough difference and we can go with whichever one.

**Reply by pml-lang** [reply]
> both now have a single-color variant.

Thank you. Seeing both variations side by side really helps to decide. Your version with two colors definitely looks better. Let's use two colors.

> If you think it helps, we can go with that.

Your version looks more modern. The alternative version mitigates the 'n or m' ambiguity. I'm ok with both versions. You did a great job, and I let you decide.

**Reply by slanden** [reply]
> You did a great job, and I let you decide.

Thanks! Oh man, choices choices.. Well, I was ready to go with the original because it felt like it flowed more and because my brain has been looking at them both too long to see a big difference.

But, for each variation I asked a separate person what they thought and the consensus was that your version with the add-on below the 'd' looked to them like a "dm", whereas the other got mixed answers. It might be better to go ahead with your version.

l sent it to you in an email so you can decide where you'd like to upload it.

**Reply by pml-lang** [reply]
Awesome !!!
Thank you very much.

**Comment by pdml-lang** [comment]
> I'll let you know when the new site is ready.

It's now ready, as announced [here](https://github.com/pdml-lang/pdml-lang.github.io/discussions/3).

> I will add a link to your plugin on the PDML and PML websites.

I added links on the [FAQ page](https://pdml-lang.dev/docs/faq/index.html#ide) and [Downloads page](https://pdml-lang.dev/downloads/index.html) of the PDML website.

**Reply by slanden** [reply]
Awesome, I updated the VS Code extension's readme with the new link.

> added links on the [FAQ page](https://pdml-lang.dev/docs/faq/index.html#ide) and [Downloads page](https://pdml-lang.dev/downloads/index.html) of the PDML website.

Thanks!

One thing I noticed with the site was when I visit, for example, https://pdml-lang.dev/docs/extensions/reference_manual/index.html, click a link on the left, then click the browser's _Back_ button, nothing happens. Might be something going wrong with the History API, but not sure.

**Reply by pdml-lang** [reply]
> I updated the VS Code extension's readme with the new link.

Nice. Thanks!

> One thing I noticed with the site was when I visit ...

Yes, I can reproduce that. Very strange. I'll have a deeper look into it.
